modules/fcrepo/main.tf

```diff
@@ -0,0 +1,150 @@
+  bucket      = "${var.S3Bucket}"
+  key         = "${var.S3Key}"
+}
+
+resource "aws_elastic_beanstalk_configuration_template" "fedora" {
+  name                = "fcrepo-template-config"
+  description         = "Fcrepo configuration template"
+  application         = "${aws_elastic_beanstalk_application.fedora.name}"
+  solution_stack_name = "64bit Amazon Linux 2017.03 v2.6.5 running Tomcat 8 Java 8"
+
+  setting {
+    namespace = "aws:autoscaling:launchconfiguration"
+    name      = "InstanceType"
+    value     = "${var.InstanceType}"
+  }
+  setting {
+    namespace = "aws:autoscaling:launchconfiguration"
+    name      = "EC2KeyName"
+    value     = "${var.KeyName}"
+  }
+  setting {
+    namespace = "aws:autoscaling:launchconfiguration"
+    name      = "SSHSourceRestriction"
+    value     = "tcp, 22, 22, 10.0.0.0/16"
+  }
+  setting {
+    namespace = "aws:autoscaling:launchconfiguration"
+    name      = "SecurityGroups"
+    value     = "${aws_security_group.fedora.id}"
+  }
+  setting {
+    namespace = "aws:autoscaling:loadbalancer"
+    name      = "SecurityGroups"
+    value     = "${aws_security_group.fedora_lb.id}"
+  }
+  setting {
+    namespace = "aws:elb:loadbalancer"
+    name      = "CrossZone"
+    value     = "true"
+  }
+  setting {
+    namespace = "aws:elb:policies"
+    name      = "ConnectionDrainingEnabled"
+    value     = "true"
+  }
+  setting {
+    namespace = "aws:ec2:vpc"
+    name      = "ELBScheme"
+    value     = "internal"
+  }
+  setting {
+    namespace = "aws:ec2:vpc"
+    name      = "Subnets"
+    value     = "${var.SubnetID}"
+  }
+  setting {
+    namespace = "aws:ec2:vpc"
+    name      = "ELBSubnets"
+    value     = "${var.SubnetID}"
+  }
+  setting {
+    namespace = "aws:autoscaling:launchconfiguration"
+    name      = "IamInstanceProfile"
+    value     = "${aws_iam_instance_profile.fedora.name}"
+  }
+  setting {
+    namespace = "aws:autoscaling:asg"
+    name      = "MinSize"
+    value     = "${var.MinSize}"
+  }
+  setting {
+    namespace = "aws:autoscaling:asg"
+    name      = "MaxSize"
+    value     = "${var.MaxSize}"
+  }
+  setting {
+    namespace = "aws:autoscaling:trigger"
+    name      = "LowerBreachScaleIncrement"
+    value     = "0"
+  }
+  setting {
+    namespace = "aws:autoscaling:trigger"
+    name      = "UpperBreachScaleIncrement"
+    value     = "0"
+  }
+  setting {
+    namespace = "aws:elasticbeanstalk:environment"
+    name      = "EnvironmentType"
+    value     = "LoadBalanced"
+  }
+  setting {
+    namespace = "aws:elasticbeanstalk:container:tomcat:jvmoptions"
+    name      = "Xmx"
+    value     = "2g"
+  }
+  setting {
+    namespace = "aws:elasticbeanstalk:container:tomcat:jvmoptions"
+    name      = "Xms"
+    value     = "2g"
+  }
+  setting {
+    namespace = "aws:elasticbeanstalk:container:tomcat:jvmoptions"
+    name      = "JVM Options"
+    value     = "${data.template_file.java_config.rendered}"
+  }
```

Comment:
Does splitting this out into a template give us anything over just listing them here?

Reply:
Just readability, which is a YMMV.  I mentioned in the other PR my feelings there on heredocs of data in the middle of 'code', but I'm good to embed it.  Especially since this is just normal lines and not formatted data.